### DISCUSSION TITLE
Mirth inserting field names

### DISCUSSION THREAD

**Discussion by Bugzy-can** [nextgenhealthcare/connect#5380] [General]
We ran into an issue recently at a customer site, where HL7 merges were arriving sometimes with a single value for the non-survivor MRN in MRG-1, but occasionally sending over a multi part field

Normal message

MRG|12345||||

Multi part

MRG|12345^^^ID~23456^^^ID||||

The first example, parsing the value works fine using msg['MRG']['MRG.1']['MRG.1.1'].toString()

However, when the multi field was sent, Merge would return

<MRG.1.1/><MRG.1.1/><MRG.1.1/>

This caused a mess with patient records. Fortunately there were not a lot of these.

The solution was to insert [0] to only look at the first element, but can someone tell me why mirth does this by default? Wouldn't it be better to return a blank rather than this garbage field data?
Is there a setting somewhere to affect this? Strict parser is not currently enabled, would that help?

**Comment by jonbartels** [comment]
https://docs.nextgen.com/bundle/Mirth_User_Guide_40/page/connect/connect/topics/c_HL7_v2_x_Data_Type_connect_ug.html 

MC is handling fields with and without repetitions. If a message is sent with only one repetition it's just handled, if its sent with more than one it's parsed and you get the behavior you described.

You should experiment with turning "Parse Field Repetitions" off and on and see what different results you get. 

I'll ask one of the moderators move this from issues to discussions. Discussions is fairly active and another expert can probably explain this better than me.

**Reply by mjtrac** [reply]
From the GUI in recent releases, you can choose to use an iterator.  Or, in Javascript, your code would be something like this, in order to put a list of the repeated values into your channel map:
```
var list_of_mrg1s = Lists.list();
for each (fld in msg['MRG']['MRG.1']){
 list_of_mrg1s.append(fld['MRG.1.1'].toString());
}
channelMap.put('mrg1values',list_of_mrg1s);
